File: tests/testdata/codegen/basic-unions-pp/python/basic-unions.py
import pulumi
import pulumi_basic_unions as basic_unions

# properties field is bound to union case ServerPropertiesForReplica
replica = basic_unions.ExampleServer("replica", properties=basic_unions.ServerPropertiesForReplicaArgs(
    create_mode="Replica",
    version="0.1.0-dev",
))
# properties field is bound to union case ServerPropertiesForRestore
restore = basic_unions.ExampleServer("restore", properties=basic_unions.ServerPropertiesForRestoreArgs(
    create_mode="PointInTimeRestore",
    restore_point_in_time="example",
))

File: pkg/codegen/dotnet/testdata/basic-unions-pp/basic-unions.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using BasicUnions = Pulumi.BasicUnions;

return await Deployment.RunAsync(() => 
{
    // properties field is bound to union case ServerPropertiesForReplica
    var replica = new BasicUnions.ExampleServer("replica", new()
    {
        Properties = new BasicUnions.Inputs.ServerPropertiesForReplicaArgs
        {
            CreateMode = "Replica",
            Version = "0.1.0-dev",
        },
    });

    // properties field is bound to union case ServerPropertiesForRestore
    var restore = new BasicUnions.ExampleServer("restore", new()
    {
        Properties = new BasicUnions.Inputs.ServerPropertiesForRestoreArgs
        {
            CreateMode = "PointInTimeRestore",
            RestorePointInTime = "example",
        },
    });

});

